sympy/sets/tests/test_fancysets.py

```diff
@@ -441,6 +441,18 @@ def test_imageset_intersect_real():
     assert s.intersect(S.Reals) == imageset(Lambda(n, 2*n*pi - pi/4), S.Integers)
 
 
+def test_imageset_intersect_interval():
```

Comment:
You don't have to make them work with Complex region, just handle them
gracefully so that the method do not give wrong result.


On Thursday 26 May 2016 07:05 PM, Kshitij Saraogi wrote:
>
> In sympy/sets/tests/test_fancysets.py
> <https://github.com/sympy/sympy/pull/11164#discussion_r64745587>:
>
> > @@ -441,6 +441,18 @@ def test_imageset_intersect_real():
> >      assert s.intersect(S.Reals) == imageset(Lambda(n, 2*n*pi - pi/4), S.Integers)
> >  
> >  
> > +def test_imageset_intersect_interval():
>
> I needed help with the case of complex functions.
> Since non-real |Interval| cannot be supported, this method won't work.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/11164/files/f6c9c016782fc4cbc2af3e721246ec7e2463e286#r64745587>
>